File: server/src/main/scala/stasis/server/api/requests/UpdateDatasetDefinition.scala
package stasis.server.api.requests

import stasis.server.model.datasets.DatasetDefinition
import stasis.server.model.schedules.Schedule

final case class UpdateDatasetDefinition(
  schedule: Option[Schedule.Id],
  redundantCopies: Int,
  existingVersions: DatasetDefinition.Retention,
  removedVersions: DatasetDefinition.Retention
)

object UpdateDatasetDefinition {
  implicit class RequestToUpdatedDefinition(request: UpdateDatasetDefinition) {
    def toUpdatedDefinition(definition: DatasetDefinition): DatasetDefinition =
      definition.copy(
        schedule = request.schedule,
        redundantCopies = request.redundantCopies,
        existingVersions = request.existingVersions,
        removedVersions = request.removedVersions
      )
  }
}

File: client-android/lib/src/main/kotlin/stasis/client_android/lib/model/server/api/requests/UpdateDatasetDefinition.kt
package stasis.client_android.lib.model.server.api.requests

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import stasis.client_android.lib.model.server.datasets.DatasetDefinition

@JsonClass(generateAdapter = true)
data class UpdateDatasetDefinition(
    val info: String,
    @field:Json(name = "redundant_copies")
    val redundantCopies: Int,
    @field:Json(name = "existing_versions")
    val existingVersions: DatasetDefinition.Retention,
    @field:Json(name = "removed_versions")
    val removedVersions: DatasetDefinition.Retention
)
